internal/component/beyla/beyla_linux.go

```diff
@@ -0,0 +1,244 @@
+	}
+}
+
+func (args Attributes) Convert() beyla.Attributes {
+	return beyla.Attributes{
+		Kubernetes: transform.KubernetesDecorator{
+			Enable: transform.KubeEnableFlag(args.Kubernetes.Enable),
+		},
+	}
+}
+
+func (args Discovery) Convert() (services.DiscoveryConfig, error) {
+	srv, err := args.Services.Convert()
+	if err != nil {
+		return services.DiscoveryConfig{}, err
+	}
+	return services.DiscoveryConfig{
+		Services: srv,
+	}, nil
+}
+
+func (args Services) Convert() (services.DefinitionCriteria, error) {
+	var attrs services.DefinitionCriteria
+	for _, s := range args {
+		ports, err := stringToPortEnum(s.OpenPorts)
+		if err != nil {
+			return nil, err
+		}
+		paths, err := stringToRegexpAttr(s.Path)
+		if err != nil {
+			return nil, err
+		}
+		attrs = append(attrs, services.Attributes{
+			Name:      s.Name,
+			Namespace: s.Namespace,
+			OpenPorts: ports,
+			Path:      paths,
+		})
+	}
+	return attrs, nil
+}
+
+func New(opts component.Options, args Arguments) (*Component, error) {
+	reg := prometheus.NewRegistry()
+	c := &Component{
+		opts:   opts,
+		args:   args,
+		reload: make(chan struct{}, 1),
+		reg:    reg,
+	}
+
+	if err := c.Update(args); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// Run implements component.Component.
+func (c *Component) Run(ctx context.Context) error {
+	var cancel context.CancelFunc
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-c.reload:
+			// cancel any previously running exporter
+			if cancel != nil {
+				cancel()
+			}
+			newCtx, cancelFunc := context.WithCancel(ctx)
+			cancel = cancelFunc
+
+			c.mut.Lock()
+			cfg, err := c.args.Convert()
+			if err != nil {
+				return fmt.Errorf("failed to convert arguments: %w", err)
+			}
+			cfg.Prometheus.Registry = c.reg
+			c.mut.Unlock()
+			components.RunBeyla(newCtx, cfg)
+		}
+	}
+}
+
+// Update implements component.Component.
+func (c *Component) Update(args component.Arguments) error {
+	c.mut.Lock()
+	baseTarget, err := c.baseTarget()
+	if err != nil {
+		return err
+	}
+	c.opts.OnStateChange(Exports{
+		Targets: []discovery.Target{baseTarget},
+	})
+	c.mut.Unlock()
+	select {
+	case c.reload <- struct{}{}:
+	default:
+	}
+	return nil
+}
+
+func (c *Component) baseTarget() (discovery.Target, error) {
+	data, err := c.opts.GetServiceData(http_service.ServiceName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get HTTP information: %w", err)
+	}
+	httpData := data.(http_service.Data)
+
+	return discovery.Target{
+		model.AddressLabel:     httpData.MemoryListenAddr,
+		model.SchemeLabel:      "http",
+		model.MetricsPathLabel: path.Join(httpData.HTTPPathForComponent(c.opts.ID), "metrics"),
+		"instance":             defaultInstance(),
+		"job":                  "beyla",
+	}, nil
+}
+
+func (c *Component) Handler() http.Handler {
+	return promhttp.HandlerFor(c.reg, promhttp.HandlerOpts{})
+}
+
+func (a *Arguments) Convert() (*beyla.Config, error) {
+	var err error
+	cfg := &beyla.DefaultConfig
```

Comment:
This is taking the address of the default config from the Beyla package, and then overwriting it with new values; this means that two `beyla.ebpf` components will have a race condition here as they fight over what the "actual" defaults are.

I'd recommend making a copy of beyla.DefaultConfig here.